### ISSUE TITLE
Warns are not summarized at the end of dbt test

### ISSUE THREAD

**Issue by MartinGuindon** [fishtown-analytics/dbt#1597]
#  Issue

When using the new _warn_ severity for tests, any warn results are not summarized at the end of the dbt test output, hence they could be easy to miss in larger projects.

### Results

The current ouput looks like everything is fine and dandy, even if there are warnings:
```
15:23:56 | Finished running 282 tests in 19.27s.

Completed successfully

Done. PASS=282 ERROR=0 SKIP=0 TOTAL=282
```

I would instead expect an output similars to errors, summarizing how many warnings there are and which test(s) are returning warnings:

```
Completed successfully with 1 warnings:

Warning in test employees_overlapping_dates (tests/employees_overlapping_dates.sql)
  Got 23 results, expected 0.

  compiled SQL at target/compiled/absobi_dbt_operations/data_test/employees_overlapping_dates.sql

Done. PASS=281 ERROR=0 WARN=1 SKIP=0 TOTAL=282
```

**Comment by drewbanin** [comment]
Thanks @MartinGuindon - really good idea! Would love to add this for the 0.14.0 bugfix/patch release

**Comment by emilieschario** [comment]
Digging into this a bit, here is what I'm finding:

* we'll need to parse the warn errors and then putting that into here https://github.com/fishtown-analytics/dbt/blob/399b33822a85493a8790df787eed59e951aecf6c/core/dbt/ui/printer.py#L330
* and update this function to reflect those changes https://github.com/fishtown-analytics/dbt/blob/399b33822a85493a8790df787eed59e951aecf6c/core/dbt/ui/printer.py#L317

From looking at the run task, it should trickle through if that is the update. 
https://github.com/fishtown-analytics/dbt/blob/399b33822a85493a8790df787eed59e951aecf6c/core/dbt/task/run.py#L14

I will have to dig deeper into how the results are parsed for errors to see how/if warns needs to be different

Am I thinking about this the right way?

**Comment by beckjake** [comment]
This is the key method: https://github.com/fishtown-analytics/dbt/blob/399b33822a85493a8790df787eed59e951aecf6c/core/dbt/ui/printer.py#L266

I think what I would do is:
 - change `TestRunner.after_execute()` [here](https://github.com/fishtown-analytics/dbt/blob/399b33822a85493a8790df787eed59e951aecf6c/core/dbt/node_runners.py#L484-L485) to call a new function you make in `dbt.ui.printer`
 - have that new function be a lot like the current `print_run_end_messages`, except have it call a new function instead of `print_run_status_line` (`print_test_status_line`?)
 - have `print_test_status_line` handle the `warn` results accordingly

**Comment by emilieschario** [comment]
Got it! Thanks for that additional color @beckjake. Could you assign this to me to indicate it's being worked on? I'd like to take a stab at it.

**Comment by drewbanin** [comment]
Hey @emilieschario - is this in progress? We're hoping to cut an RC of 0.14.1 early next week, then publish the final 0.14.1 release later in the week. Let me know if there's anything we can help out with!

**Comment by emilieschario** [comment]
Sorry @drewbanin I won't have the bandwidth to get to this quickly. No worries if you'd rather reassign this! 

In the future, I'll make sure I have time carved out before asking for it to be assigned!

**Comment by drewbanin** [comment]
No problem at all @emilieschario! Thanks for the update :)

**Comment by drewbanin** [comment]
Closed by #1654
